### PR TITLE
test: Exclude sys/sysmacros.h on FreeBSD

### PR DESCRIPTION
**Pull request by diizzyy**
Fixes build issue on FreeBSD

### CODE REVIEW

File: test/mknod_test.c [hunk 1]
```diff
@@ -28,8 +28,11 @@
 #include <string.h>
 #include <errno.h>
 #ifndef __APPLE__
+#elif
+#ifndef __FreeBSD__
```
Comment (gaul) [inline test/mknod_test.c:32]: I am skeptical of code that CI does not exercise which will rot over time.  Is there some other way to approach this?

Comment (diizzyy) [reply test/mknod_test.c:32]: I'm not sure what you mean, that file is Linux-kernel specific.

Comment (gaul) [reply test/mknod_test.c:32]: Could we call `system` to invoke `mknod` instead?

Comment (gaul) [reply test/mknod_test.c:32]: Thinking about this some more I think the `#ifdef` is fine.  But CI is broken so we cannot merge this.

Comment (gaul) [reply test/mknod_test.c:32]: @diizzyy Could you move forward with this PR?

Comment (diizzyy) [reply test/mknod_test.c:32]: I'll try to have a look again next week, unfortunately I don't have a Linux box available so I'm not sure why it fails.

**Comment by gaul** [comment]
Closing this due to inactivity.  Please reopen if still relevant.  If we can improve FreeBSD support I would like to do that and especially get its packages on the latest version.

**Pull request closed without merging**